src/bootstrap/lib.rs

```diff
@@ -426,7 +426,8 @@ impl Build {
         // we always try to use git for LLVM builds
         let in_tree_llvm_info = channel::GitInfo::new(false, &src.join("src/llvm-project"));
 
-        let initial_target_libdir_str = if config.dry_run {
+        let initial_target_libdir_str = if config.dry_run || config.initial_rustc == PathBuf::new()
```

Comment:
I think this is caused by the Build::new construction I commented on before, which is why I was suspicious of it. :)